src/main/kotlin/org/rust/lang/core/resolve/ref/RsPathReferenceImpl.kt

```diff
@@ -80,41 +79,44 @@ class RsPathReferenceImpl(
     }
 
     override fun advancedResolve(): BoundElement<RsElement>? {
-        val resolved = advancedMultiResolve().singleOrNull() ?: return null
-        return instantiatePathGenerics(element, resolved.element, resolved.resolvedSubst)
+        val resolved = rawMultiResolve().singleOrNull() ?: return null
+        return TyLowering.lowerPathGenerics(
```

Comment:
Why don't we pass `resolvedNestedPaths` to `TyLowering` here? Can't we somehow reuse result of `resolveAllNestedPathsInternal` from `rawCachedMultiResolve`

Reply:
Good catch! Fixed now